### PR TITLE
(onionshare) Fix detection of new version

### PR DESCRIPTION
**Pull request by TakataSanshiro**
And I replaced the old icon.

But, an error displayed when I run this update.ps1.
Running au_BeforeUpdate
Get-ValidOpenSourceLicense : The term 'Get-ValidOpenSourceLicense' is not recognized as the name of a cmdlet,
function, script file, or operable program.

### CODE REVIEW

**Comment by AdmiringWorm** [comment]
> Get-ValidOpenSourceLicense : The term 'Get-ValidOpenSourceLicense' is not recognized as the name of a cmdlet,
function, script file, or operable program.

That is because the cmdlet is only imported automatically when running a full update of all packages.
When testing a single package, either use the `test_all.ps1` script and the package name as an argument, or you need to import the file `./scripts/au_extensions.psm1`.

Anyways.
As you can see from the failure on the build, just updating the `update.ps1` script is not enough, unfortunately. For me to accept this PR, the install/uninstall arguments passed to the installer need to be updated, the unit tests need to be updated and the autohotkey script need to be updated (or removed if it is no longer necessary).

**Comment by TakataSanshiro** [comment]
Import the au_extensions.psm1 was worked.

File: automatic/onionshare/tools/chocolateyUninstall.ps1 [hunk 1]
```diff
@@ -1,25 +1,35 @@
...
 
 $toolsDir = Split-Path -Parent $MyInvocation.MyCommand.Definition
-$packageArgs = @{
-  packageName    = $env:ChocolateyPackageName
-  softwareName   = 'OnionShare'
-  fileType       = 'exe'
-  silentArgs     = '/S'
-  validExitCodes = @(0)
-}
+$packageName    = $env:ChocolateyPackageName
```
Comment (AdmiringWorm) [inline automatic/onionshare/tools/chocolateyUninstall.ps1:4]: Please don't make unneccessary changes, the variables should not be overused and be in the `$packageArgs` hashtable.

File: automatic/onionshare/tools/chocolateyUninstall.ps1 [hunk 2]
```diff
@@ -1,25 +1,35 @@
...
+$softwareName   = 'OnionShare'
+$installerType  = 'msi'
+$silentArgs     = '/qn /norestart'
+$validExitCodes = @(0)
 
 $uninstalled = $false
 
-[array]$key = Get-UninstallRegistryKey @packageArgs
+[array]$key = Get-UninstallRegistryKey -SoftwareName $softwareName
```
Comment (AdmiringWorm) [inline automatic/onionshare/tools/chocolateyUninstall.ps1:12]: Why are you changing this, please change it back to how it was

File: automatic/onionshare/tools/chocolateyUninstall.ps1 [hunk 3]
```diff
@@ -1,25 +1,35 @@
...
 
 if ($key.Count -eq 1) {
-  $key | ForEach-Object {
-    $packageArgs['file'] = "$($_.UninstallString)"
+  $file = "$($key.UninstallString)"
```
Comment (AdmiringWorm) [inline automatic/onionshare/tools/chocolateyUninstall.ps1:15]: This is completely unneccessary when using an `MSI`. Please remove the line, and keep attention to spacing (there should be 4 spaces at this location).

File: automatic/onionshare/tools/chocolateyUninstall.ps1 [hunk 4]
```diff
@@ -1,25 +1,35 @@
...
+
+  # The Product Code GUID is all that should be passed for MSI, and very
+  # FIRST, because it comes directly after /x, which is already set in the
+  # Uninstall-ChocolateyPackage msiargs (facepalm).
+  $silentArgs = "$($key.PSChildName) $silentArgs"
+
+  # Don't pass anything for file, it is ignored for msi (facepalm number 2)
+  # Alternatively if you need to pass a path to an msi, determine that and
+  # use it instead of the above in silentArgs, still very first.
+  $file = ''
```
Comment (AdmiringWorm) [inline automatic/onionshare/tools/chocolateyUninstall.ps1:25]: This can be moved to the `$packageArgs` instead. Please also remove the comments as these are not needed.

File: automatic/onionshare/tools/chocolateyUninstall.ps1 [hunk 5]
```diff
@@ -1,25 +1,35 @@
...
 
-    Uninstall-ChocolateyPackage @packageArgs
-    Start-Process -Wait "autohotkey" -ArgumentList "$toolsDir\uninstall.ahk"
-  }
+  Uninstall-ChocolateyPackage `
```
Comment (AdmiringWorm) [inline automatic/onionshare/tools/chocolateyUninstall.ps1:27]: Don't change the call to `Uninstall-ChocolateyPackage`, it should use the variables set in `packageArgs`, not seperate variables that only pollutes the current process.

File: automatic/onionshare/onionshare.nuspec [hunk 6]
```diff
@@ -3,14 +3,14 @@
 <package xmlns="http://schemas.microsoft.com/packaging/2015/06/nuspec.xsd">
   <metadata>
     <id>onionshare</id>
-    <version>2.2</version>
+    <version>2.3.1</version>
```
Comment (AdmiringWorm) [inline automatic/onionshare/onionshare.nuspec:6]: Do NOT update the version, you are effectively preventing automatic updates now.

File: automatic/onionshare/onionshare.nuspec [hunk 7]
```diff
@@ -3,14 +3,14 @@
...
     <packageSourceUrl>https://github.com/AdmiringWorm/chocolatey-packages/tree/master/automatic/onionshare</packageSourceUrl>
     <owners>AdmiringWorm</owners>
     <title>OnionShare</title>
     <authors>Micah Lee</authors>
     <projectUrl>https://onionshare.org/</projectUrl>
     <iconUrl>https://cdn.jsdelivr.net/gh/AdmiringWorm/chocolatey-packages@92f4e01bb66a1a5a853e465e8a06ae7017de871a/automatic/onionshare/icons/128x128.png</iconUrl>
-    <licenseUrl>https://github.com/micahflee/onionshare/blob/v2.2/LICENSE</licenseUrl>
+    <licenseUrl>https://github.com/micahflee/onionshare/blob/v2.3.1/LICENSE</licenseUrl>
```
Comment (AdmiringWorm) [inline automatic/onionshare/onionshare.nuspec:13]: This should be set by the update script, there is no need to set it manually. Please revert the change.

File: automatic/onionshare/tools/chocolateyInstall.ps1 [hunk 8]
```diff
@@ -4,13 +4,13 @@ $toolsPath = Split-Path -parent $MyInvocation.MyCommand.Definition
 
 $packageArgs = @{
   packageName    = $env:ChocolateyPackageName
-  fileType       = 'exe'
-  file           = "$toolsPath\onionshare-2.2-setup.exe"
+  fileType       = 'msi'
+  file           = "$toolsPath\OnionShare-2.3.1.msi"
   softwareName   = 'OnionShare'
-  silentArgs     = '/S'
+  silentArgs     = '/quiet /passive /norestart'
```
Comment (AdmiringWorm) [inline automatic/onionshare/tools/chocolateyInstall.ps1:10]: What about the log? Typically MSI's allows a log path to be specified as well, and when possible it should be set.

**Comment by TakataSanshiro** [comment]
I changed the uninstall script then working fine.
I confirmed that the autohotkey script is no longer needed.
This original uninstall script isn't working. I wasn't able to fix this and uninstall the previous one.

**Comment by AdmiringWorm** [comment]
Your changes has been merged into a separate branch, unfortunately there is still a bunch of work that needs to be done before I can push any new version upstream (most is now completed, but there is still some left).

**Comment by AdmiringWorm** [comment]
Changes to the package, including the changes you provided are now available in the master branch and should be pushed to chocolatey.org shortly.

Thank you for your contribution :+1:

**Comment by AdmiringWorm** [comment]
@allcontributors-bot please add @TakataSanshiro for code

**Comment by AdmiringWorm** [comment]
@all-contributors please add @TakataSanshiro for code
